spring-boot-project/spring-boot/src/main/java/org/springframework/boot/context/properties/PropertyMapper.java

```diff
@@ -308,6 +309,23 @@ public <R> R toInstance(Function<T, R> factory) {
 			return factory.apply(value);
 		}
 
+		/**
+		 * Complete the mapping by creating a new instance from the non-filtered value or
+		 * using the specified default instance when the value is filtered.
+		 * @param <R> the resulting type
+		 * @param factory the factory used to create the instance
+		 * @param defaultInstance the instance to use if the value has been filtered
+		 * @return the instance
+		 */
+		public <R> R toInstance(Function<T, R> factory, R defaultInstance) {
```

Comment:
I went w/ a direct value for the default rather than a supplier of the default. Mostly because that is the mode that works for my current need. 

However, I do see that for the `from` API we have both supplier and direct value version of the API. Should we do the same here, or keep one or the other?